code/game/machinery/portable_turret.dm

```diff
@@ -362,40 +225,40 @@ Status: []<BR>"},
 			// try and salvage its components
 			to_chat(user, "You begin prying the metal coverings off.")
 			sleep(20)
-			if(prob(70))
-				to_chat(user, "You remove the turret and salvage some components.")
-				if(installation)
-					var/obj/item/weapon/gun/energy/Gun = new installation(src.loc)
-					Gun.power_supply.charge=gun_charge
-					Gun.update_icon()
+			var/salvaged
+			if(installed)
+				if(prob(70))
+					var/obj/item/I = installed
+					I.forceMove(get_turf(src))
+					installed = null
 					lasercolor = null
-				if(prob(50))
-					var/obj/item/stack/sheet/metal/M = getFromPool(/obj/item/stack/sheet/metal,loc)
-					M.amount = rand(1,4)
-				if(prob(50))
-					new /obj/item/device/assembly/prox_sensor(locate(x,y,z))
+					salvaged++
+			if(prob(75))
+				var/obj/item/stack/sheet/metal/M = getFromPool(/obj/item/stack/sheet/metal,loc)
+				M.amount = rand(2,6)
+				salvaged++
+			if(prob(50))
+				new /obj/item/device/assembly/prox_sensor(locate(x,y,z))
+				salvaged++
+			if(salvaged)
+				to_chat(user, "You remove the turret and salvage some components.")
 			else
 				to_chat(user, "You remove the turret but did not manage to salvage anything.")
 			qdel(src)
 		return
 
 	..()
 
-	if((iswrench(W)) && (!on))
-		if(raised)
-			return
+	if((iswrench(W)) && (!on) && !raised && wrenchAnchor(user))
```

Comment:
Don't need to put parentheses around individual expressions like `iswrench()` like this.